socorro/unittest/external/boto/test_connection_context.py

```diff
@@ -218,3 +226,294 @@ def test_fetch_with_regional_s3connection_context(self):
             'us-south-3',
             connection_source
         )
+
+
+
+class MultiplePathsBase(socorro.unittest.testbase.TestCase):
+    """Sets up mocked s3 storage using DatePrefixKeyBuilder"""
+    def setup_mocked_s3_storage(
+        self,
+        executor=TransactionExecutor,
+        executor_for_gets=TransactionExecutor,
+        storage_class='BotoS3CrashStorage',
+        host='',
+        port=0,
+        resource_class=S3ConnectionContext,
+        **extra
+    ):
+        config = DotDict({
+            'resource_class': resource_class,
+            'logger': mock.Mock(),
+            'host': host,
+            'port': port,
+            'access_key': 'this is the access key',
+            'secret_access_key': 'secrets',
+            'bucket_name': 'silliness',
+            'keybuilder_class': DatePrefixKeyBuilder,
+            'prefix': 'dev',
+            'calling_format': mock.Mock()
+        })
+        config.update(extra)
+        s3_conn = resource_class(config)
+        s3_conn._connect_to_endpoint = mock.Mock()
+        s3_conn._mocked_connection = s3_conn._connect_to_endpoint.return_value
+        s3_conn._calling_format.return_value = mock.Mock()
+        s3_conn._CreateError = mock.Mock()
+        s3_conn.ResponseError = mock.Mock()
+        s3_conn._open = mock.MagicMock()
+
+        return s3_conn
+
+
+class DatePrefixKeyBuilderTestCase(MultiplePathsBase):
+    """Tests the DatePrefixKeyBuilder with different crash types"""
+    def test_dir_builder_with_dump(self):
+        connection_source = self.setup_mocked_s3_storage()
+        prefix = 'dev'
+        name_of_thing = 'dump'
+        crash_id = 'fff13cf0-5671-4496-ab89-47a922141114'
+        all_keys = connection_source.build_keys(prefix, name_of_thing, crash_id)
+
+        # Only the old-style key is returned
+        self.assertEqual(len(all_keys), 1)
+        self.assertEqual(
+            all_keys[0],
+            'dev/v1/dump/fff13cf0-5671-4496-ab89-47a922141114'
+        )
+
+    def test_dir_builder_with_raw_crash(self):
+        connection_source = self.setup_mocked_s3_storage()
+        prefix = 'dev'
+        name_of_thing = 'raw_crash'
+        crash_id = 'fff13cf0-5671-4496-ab89-47a922141114'
+        all_keys = connection_source.build_keys(prefix, name_of_thing, crash_id)
+
+        # The new- and old-style keys are returned
+        self.assertEqual(len(all_keys), 2)
+        self.assertEqual(
+            all_keys[0],
+            'dev/v2/raw_crash/fff/20141114/fff13cf0-5671-4496-ab89-47a922141114'
```

Comment:
It's not impossible. You'd have to try all `16 ^ 3` combinations. This is easy enough to script though. You can see how the GETs to retrieve a day scale here: https://www.wolframalpha.com/input/?i=table+30+*+(16+%5E+z)+from+0+to+4